sw/device/tests/kmac_error_conditions_test.c

```diff
@@ -0,0 +1,583 @@
+                       0x53525150, 0x57565554, 0x5b5a5958, 0x5f5e5d5c},
+            .share1 = {0},
+            .length = kDifKmacKeyLen256,
+        },
+    .message =
+        "\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f"
+        "\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f"
+        "\x20\x21\x22\x23\x24\x25\x26\x27\x28\x29\x2a\x2b\x2c\x2d\x2e\x2f"
+        "\x30\x31\x32\x33\x34\x35\x36\x37\x38\x39\x3a\x3b\x3c\x3d\x3e\x3f"
+        "\x40\x41\x42\x43\x44\x45\x46\x47\x48\x49\x4a\x4b\x4c\x4d\x4e\x4f"
+        "\x50\x51\x52\x53\x54\x55\x56\x57\x58\x59\x5a\x5b\x5c\x5d\x5e\x5f"
+        "\x60\x61\x62\x63\x64\x65\x66\x67\x68\x69\x6a\x6b\x6c\x6d\x6e\x6f"
+        "\x70\x71\x72\x73\x74\x75\x76\x77\x78\x79\x7a\x7b\x7c\x7d\x7e\x7f"
+        "\x80\x81\x82\x83\x84\x85\x86\x87\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f"
+        "\x90\x91\x92\x93\x94\x95\x96\x97\x98\x99\x9a\x9b\x9c\x9d\x9e\x9f"
+        "\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa9\xaa\xab\xac\xad\xae\xaf"
+        "\xb0\xb1\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\xbb\xbc\xbd\xbe\xbf"
+        "\xc0\xc1\xc2\xc3\xc4\xc5\xc6\xc7",
+    .message_len = 200,
+    .customization_string = "My Tagged Application",
+    .customization_string_len = 21,
+    .digest = {0x1c73bed5, 0x73d74e95, 0x59bb4628, 0xe3a8e3db, 0x7ae7830f,
+               0x5944ff4b, 0xb4c2f1f2, 0xceb8ebec, 0xc601ba67, 0x57b88a2e,
+               0x9b492d8d, 0x6727bbd1, 0x90117868, 0x6a300a02, 0x1d28de97,
+               0x5d3030cc},
+    .digest_len = 16,
+    .digest_len_is_fixed = false,
+};
+
+/**
+ * Init key manager.
+ *
+ * @return OK or error.
+ */
+status_t setup_keymgr(void) {
+  // Initialize keymgr and advance to the state specified by the loaded ROM.
+  CHECK_STATUS_OK(keymgr_testutils_initialize(&keymgr, &kmac));
+
+  return OK_STATUS();
+}
+
+/**
+ * Poll the SHA3_IDLE bit.
+ *
+ * @param kmac The KMAC handle.
+ */
+void poll_sha3_idle_bit(const dif_kmac_t *kmac) {
+  uint32_t reg;
+  do {
+    reg = mmio_region_read32(kmac->base_addr, KMAC_STATUS_REG_OFFSET);
+  } while (!bitfield_bit32_read(reg, KMAC_STATUS_SHA3_IDLE_BIT));
+}
+
+/**
+ * Check and clear the KMAC error.
+ *
+ * Performs the following:
+ * - Check whether error interrupt was seen.
+ * - Check whether the expected error was seen.
+ * - Clears the error interrupt.
+ * - Sets the error processed bit.
+ * - Reset the KMAC IP.
+ * - Waits until KMAc again gets ready.
+ *
+ * @param kmac The KMAC handle.
+ * @param exp_err The expected error.
+ * @return OK or error.
+ */
+status_t check_and_clear_error(const dif_kmac_t *kmac, dif_kmac_error_t exp_err,
+                               bool wait_idle) {
+  dif_kmac_error_t err;
+  dif_kmac_operation_state_t kmac_operation_state;
+  uint32_t info;
+  bool error;
+
+  // Check whether the KMAC ERR bit was set in the interrupt state register.
+  TRY(dif_kmac_has_error_occurred(kmac, &error));
+  CHECK(error, "No error was triggered.");
```

Comment:
This applies throughout this file.
```suggestion
  TRY_CHECK(error, "No error was triggered.");
```

Reply:
As far as I see, the difference between `CHECK` and `TRY_CHECK` is that the former aborts the test and the latter continues but returns a `status_t`? Wouldn't it be better to abort the test (so using `CHECK`)?

Reply:
Makes sense. Thanks for the explanation! Changed accordingly.